src/loaders/compute_tools/tool_common.py

```diff
@@ -246,46 +226,67 @@ def _get_data_ids(self):
             data_ids = all_data_ids
         return list(set(data_ids))
 
+    def _prepare_execution(
+            self,
+            unzip: bool
+    ) -> Tuple[Path, Dict[str, Dict[str, Union[str, Path]]], List[str]]:
```

Comment:
Is there any reason genomes_meta can't just always be a dict[str, dict[str, Path]]

Reply:
genomes_meta can have `tool_identifier` str field.

Reply:
In which cases does it have tool id and which cases a path?

Reply:
it will always have a mix of strings (`tool_identifier`) and Path (`data_file`)
https://github.com/kbase/collections/blob/develop/src/loaders/compute_tools/tool_common.py#L520-L523

Reply:
:+1: